### PR TITLE
Hide type models for Ruby by default

### PR DESCRIPTION
**Pull request by koesie10**
This is a first step into making type models hidden for non-canary users. To test this, you'll need to change `extensions/ql-vscode/src/model-editor/languages/ruby/index.ts` to make `isHidden: () => true` since you can't open the Ruby model editor without canary.

Currently, you will still see that there are unsaved models when the type models get generated on start up. This will be fixed in a follow-up PR.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Comment by charisk** [comment]
Does it make sense to separate what we show in the UI vs what we store in memory/on-disk? It seems a bit strange to be going so far down the stack before deciding something is hidden + be passing the canary flag all the way down.

**Comment by charisk** [comment]
Ok thanks. I think this is okay considering it's a temporary thing as well (I'm assuming!).

**Review by charisk — APPROVED** [review]

**Comment by koesie10** [comment]
> Ok thanks. I think this is okay considering it's a temporary thing as well (I'm assuming!).

I don't think this will be temporary: the intention is to permanently hide type models from users and always generate them. I would like to find a better solution for this, but couldn't really come up with one. An alternative would be to pass in something like `hiddenPredicates` into the view state and pass this down, but that would just change what data is passed down.

**Comment by charisk** [comment]
> > Ok thanks. I think this is okay considering it's a temporary thing as well (I'm assuming!).
> 
> I don't think this will be temporary: the intention is to permanently hide type models from users and always generate them. I would like to find a better solution for this, but couldn't really come up with one. An alternative would be to pass in something like `hiddenPredicates` into the view state and pass this down, but that would just change what data is passed down.

The canary part is temporary though right?

**Comment by koesie10** [comment]
> The canary part is temporary though right?

No, the intention is for users running in canary mode to be able to add and edit type models, and regular users not in canary mode should not need to know/care about type models. So, in non-canary mode we will automatically generate type models to a separate file and not allow users to view/edit/add type models, and in canary mode we will not generate type models automatically and allow users to view/edit/add type models. We might move this to a separate setting in the future rather than basing it on canary mode, but I don't think that would fundamentally change that we need to pass through the value of a setting.

**Comment by charisk** [comment]
> > The canary part is temporary though right?
> 
> No, the intention is for users running in canary mode to be able to add and edit type models, and regular users not in canary mode should not need to know/care about type models. So, in non-canary mode we will automatically generate type models to a separate file and not allow users to view/edit/add type models, and in canary mode we will not generate type models automatically and allow users to view/edit/add type models. We might move this to a separate setting in the future rather than basing it on canary mode, but I don't think that would fundamentally change that we need to pass through the value of a setting.

This does feel like a separate setting to me then - we want to have a way to flip between automatically generating types and not, for different users. I see canary as a way to gradually ship a feature to users, not to control the feature set long-term. 

Based on that, I would have `shouldGenerateTypeModels` setting (or something like that) and pass that in the React views rather than `isCanary`. For now we could control its value based on `isCanary`, but ideally we'd move to a more proper setting eventually. Appreciate this doesn't remove the need to prop drill. 

That's just an opinion though and I'm happy to be convinced otherwise, or even go with what you've got for now, if it's blocking you.

**Comment by koesie10** [comment]
@charisk Thank! I definitely agree, so I'll move this to `codeQL.model.showTypeModels`.

File: extensions/ql-vscode/src/view/model-editor/__tests__/ModelTypeDropdown.spec.tsx [hunk 1]
```diff
@@ -34,7 +36,7 @@ describe(ModelTypeDropdown.name, () => {
     );
   });
 
-  it("allows changing the type to 'Type' for Ruby", async () => {
+  it("allows changing the type to 'Type' for Ruby in canary mode", async () => {
```
Comment (charisk) [inline extensions/ql-vscode/src/view/model-editor/__tests__/ModelTypeDropdown.spec.tsx:39]: This needs changing now that we've stopped using canary checks. Same with the following test.

**Review by charisk — APPROVED** [review]
LGTM!

**Pull request merged**